backend/capellacollab/projects/toolmodels/modelsources/git/handler/handler.py

```diff
@@ -60,26 +62,65 @@ def get_artifact_from_job_as_content(
 
     @abc.abstractmethod
     async def get_file_from_repository(
-        self, project_id: str, trusted_file_path: str
+        self,
+        project_id: str,
+        trusted_file_path: str,
+        revision: str | None = None,
     ) -> bytes:
         pass
 
+    @abc.abstractmethod
+    def get_last_commit_id_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> t.Any:
+        pass
+
+    async def get_file_from_repository_or_artifacts_as_json(
+        self,
+        trusted_file_path: str,
+        job_name: str | None,
+        revision: str | None = None,
+    ) -> tuple[t.Any, dict]:
+        (
+            last_updated,
+            result,
+        ) = await self.get_file_from_repository_or_artifacts(
+            trusted_file_path, job_name, revision
+        )
+        return (last_updated, json.loads(result.decode("utf-8")))
+
     async def get_file_from_repository_or_artifacts(
-        self, trusted_file_path: str, job_name: str | None
-    ) -> bytes:
+        self,
+        trusted_file_path: str,
+        job_name: str | None,
+        revision: str | None = None,
+    ) -> tuple[t.Any, bytes]:
         project_id = await self.get_project_id_by_git_url()
         try:
-            return await self.get_file_from_repository(
-                project_id, trusted_file_path
+            return (
+                self.get_last_commit_id_for_file_path(
+                    project_id,
+                    trusted_file_path,
+                    revision=revision,
+                ),
+                await self.get_file_from_repository(
+                    project_id, trusted_file_path, revision
+                ),
             )
         except (requests.HTTPError, exceptions.GitRepositoryFileNotFoundError):
             pass
         if job_name:
-            (_, (job_id, _),) = await self.get_last_job_run_id_for_git_model(
+            (
+                _,
+                (job_id, last_updated),
+            ) = await self.get_last_job_run_id_for_git_model(
```

Comment:
`get_last_job_run_id_for_git_model` gets the project_id as input parameter and returns it again. This doesn't make sense and the returned project_id is also not used.